packages/core/content-releases/admin/src/components/ReleaseActionMenu.tsx

```diff
@@ -115,11 +148,47 @@ export const ReleaseActionMenu = ({
           Refactor this once fixed in the DS
          */}
         <Menu.Content top={1} popoverPlacement="bottom-end">
+          {/*
+            We show edit entry option only if contentTypeUid is provided
+            which means that we are on the Release Details page
+            and not in the EditView Injected component for releases
+          */}
```

Comment:
Instead of handling this using a condition based on a prop and adding more props why don't we use composition by exporting the StyledMenuItem. So here you would have children and then handle the update in the Release DetailsPage. Delete can stay since it exists in all places the component is currently being used.

Reply:
Since you have a lot of PRs open and you are working on the validation I can take over this one to explore this idea